### PR TITLE
Added files for SPFx starter workflow

### PR DESCRIPTION
**Pull request by anoopt**
### SharePoint Framework (SPFx) starter workflow

This PR contains a starter workflow related to projects built using SharePoint Framework (SPFx). The workflow contains steps to build the project and create the package (.sppkg file) in the artifacts folder. Once there are actions related to deploying the package to SharePoint, the plan is submit a new PR with the same. 

The following are satisfied as part of the PR.

- [x] Include a good description of the workflow.
- [ ] Links to the language or tool will be nice (unless its really obvious)

In the workflow and properties files:

- [x] Includes a matching `ci/properties/*.properties.json` file.
- [x] Use title case for the names of workflows and steps, for example "Run tests".
- [x] The name of CI workflows should only be the name of the language or platform: for example "Go" (not "Go CI" or "Go Build")
- [x] Include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] CI workflows should run `push`.
- [ ] Packaging workflows should run on `release` with `types: [created]`.

Some general notes:

- [x] Does not use an Action that isn't in the `actions` organization.
- [x] Does not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Does not use a paid service or product.

### CODE REVIEW

File: ci/spfx.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# To configure this workflow:
+#
+# Change the value of the environment variable SPPKG_FILE_NAME (below).
+
+name: SharePoint Framework (SPFx)
+
+on:
+  push:
+    branches:
+      - master
```
[the remark this answers is not in the document]
Comment (anoopt) [reply ci/spfx.yml:12]: Thank you @andymckay . Sorry I do not think there a logo for SharePoint framework. So we have to go with the SharePoint logo. If not I can crop the SPFx part from [this image](https://repository-images.githubusercontent.com/135309172/e0309700-e478-11e9-9b6c-29c477fc3817).

[the remark this answers is not in the document]
Comment (andymckay) [reply ci/spfx.yml:12]: We still need to resolve the legal issue for that logo regardless I'm afraid.

[the remark this answers is not in the document]
Comment (anoopt) [reply ci/spfx.yml:12]: Ah ok. If there is anything that needs to be done from my side, please let me know.

[the remark this answers is not in the document]
Comment (andymckay) [reply ci/spfx.yml:12]: Do you know who to contact about the logo? Would you consider using a different logo, one that has a clear license?

[the remark this answers is not in the document]
Comment (anoopt) [reply ci/spfx.yml:12]: I can add the [PnP SPFx logo](https://twitter.com/pnpspfx/photo) as in the future this starter workflow can be extended to include these [PnP GitHub Actions](https://github.com/marketplace?query=microsoft+365)

**Pull request closed without merging**